### PR TITLE
SRVCOM-1728: Update serverless abstracts to align with Jupiter requirements

### PR DESCRIPTION
**Pull request by abrennan89**
Jira: https://issues.redhat.com/browse/SRVCOM-1728

Applies for versions OCP 4.6+

Direct previews:
- https://deploy-preview-43288--osdocs.netlify.app/openshift-enterprise/latest/serverless/serverless-release-notes.html
- https://deploy-preview-43288--osdocs.netlify.app/openshift-enterprise/latest/serverless/discover/about-serverless.html
- https://deploy-preview-43288--osdocs.netlify.app/openshift-enterprise/latest/serverless/discover/serverless-functions-about.html
- https://deploy-preview-43288--osdocs.netlify.app/openshift-enterprise/latest/serverless/install/install-serverless-operator.html

### CODE REVIEW

**Review by lance — COMMENTED** [review]
/lgtm

**Comment by nak3** [comment]
LGTM!

**Review by pierDipi — APPROVED** [review]
/lgtm

**Comment by maschmid** [comment]
QE approved

File: modules/serverless-api-versions.adoc [hunk 1]
```diff
@@ -6,8 +6,8 @@
 [id="serverless-api-versions_{context}"]
 = About API versions
 
-The {ServerlessOperatorName} automatically upgrades older resources that use deprecated versions of APIs to use the latest version.
+API versions are an important measure of the development status of certain features and custom resources in {ServerlessProductName}. Creating resources on your cluster that do not use the correct API version can cause issues in your deployment.
```
Comment (sagidlow) [inline modules/serverless-api-versions.adoc:9]:
```suggestion
API versions are an important measure of the development status of certain features and custom resources in {ServerlessProductName}. Creating resources on your cluster that does not use the correct API version can cause issues in your deployment.
```

File: serverless/discover/serverless-functions-about.adoc [hunk 2]
```diff
@@ -7,13 +7,11 @@ include::_attributes/serverless-document-attributes.adoc[]
 
 toc::[]
 
+{FunctionsProductName} enables developers to create and deploy stateless, event-driven functions as a Knative service on {product-title}. The `kn func` CLI is provided as a plug-in for the Knative `kn` CLI. {FunctionsProductName} uses the link:https://buildpacks.io/[CNCF Buildpack API] to create container images. After a container image has been created, you can use the `kn func` CLI to deploy the container image as a Knative service on the cluster.
```
Comment (sagidlow) [inline serverless/discover/serverless-functions-about.adoc:10]:
```suggestion
{FunctionsProductName} enables developers to create and deploy stateless, event-driven functions as a Knative service on {product-title}. The `kn func` CLI is provided as a plug-in for the Knative `kn` CLI. {FunctionsProductName} uses the link:https://buildpacks.io/[CNCF Buildpack API] to create container images. After a container image is created, you can use the `kn func` CLI to deploy the container image as a Knative service on the cluster.
```

File: serverless/serverless-release-notes.adoc [hunk 3]
```diff
@@ -7,7 +7,7 @@ include::_attributes/serverless-document-attributes.adoc[]
 
 toc::[]
 
-This topic includes information about new features, changes, and known issues that pertain to the latest {ServerlessProductName} release on {product-title}.
+Release notes contain information about new and deprecated features, breaking changes, and known issues. The following release notes apply for the most recent {ServerlessProductName} releases on {product-title}.
```
Comment (sagidlow) [inline serverless/serverless-release-notes.adoc:10]: Breaking changes? not sure what that means.

**Review by sagidlow — COMMENTED** [review]
Just a couple of nit items. Please remember to squash your commits before merging :)

File: serverless/serverless-release-notes.adoc (hunk 3 above)
Comment (abrennan89) [reply serverless/serverless-release-notes.adoc:10]: It's a pretty common term that most developers would understand https://en.wiktionary.org/wiki/breaking_change

Comment (sagidlow) [reply serverless/serverless-release-notes.adoc:10]: ACK 👍 Thanks for explaining :)

**Pull request merged**